iroh/src/endpoint.rs

```diff
@@ -1282,7 +1307,44 @@ pub struct Connecting {
 }
 
 impl Connecting {
-    /// Convert into a 0-RTT or 0.5-RTT connection at the cost of weakened security.
+    /// Converts this [`Connecting`] into a 0-RTT or 0.5-RTT connection at the cost of weakened
+    /// security.
+    ///
+    /// Returns `Ok` immediately if the local endpoint is able to attempt sending 0/0.5-RTT data.
+    /// If so, the returned [`Connection`] can be used to send application data without waiting for
+    /// the rest of the handshake to complete, at the cost of weakened cryptographic security
+    /// guarantees. The returned [`ZeroRttAccepted`] future resolves when the handshake does
+    /// complete, at which point subsequently opened streams and written data will have full
+    /// cryptographic protection.
```

Comment:
So the `Connection` object you were already using for 0-rtt will suddenly become 1-rtt?  I think you could make this a little bit more explicit.

Reply:
Errr. There's no "becoming 1-rtt". You always need 1 RTT to do a full handshake. Using 0-RTT data simply means you pre-load the handshake with data. Once the handshake finishes, you're back to a normal connection. 0-RTT, 0.5-RTT and 1-RTT only relate to differences in how the connection handshake starts out.

Can you help me come up with docs that would've helped here?

Reply:
I'm still not sure on the semantics.  "The returned [`ZeroRttAccepted`] future resolves when the handshake does complete, at which point subsequently opened streams and written data will have full cryptographic protection."

- The `Connection` object stays the same: a stream created before `ZeroRttAccepted` completed is 0-rtt, any stream created after a `ZeroRttAccepted` is not 0-rtt.
- "and written data will have full cryptographic protection" is confusing to me.  Do streams opened before `ZeroRttAccepted` suddently become fully protected?  Or does this only refer to data written into streams that were created after `ZeroRttAccepted` completed.

Reply:
>  Do streams opened before ZeroRttAccepted suddently become fully protected?
Yes, but I suspect it's not really something one can depend on - given you don't really know how much of the stream was read in 0-RTT and how much was read after the full handshake. So it's safe to say streams opened before aren't fully protected.